Automattic/wp-calypso
# Untangling: Allow different design between Settings -> Site and Hosting -> Site Settings

Related to p1730433780974949-slack-CRWCHQGUB

## Proposed Changes

This is an attempt to refactor the render logic so that the same form component logic in Settings -> Site and Hosting -> Site Settings (Settings -> General) can be shared between the pages. Here, we introduce a new `<PanelSection />` component, which is intended to be reused in all "cards" in the new IA tab contents.

This is a preparation step. Later, when a design is ready, we can tweak the CSS of that component, without touching the old design in Hosting -> Site Settings (Settings -> General).

cc: @lucasmendes-design 

|Settings -> Site|Hosting -> Site Settings<br>(`flags=-untangling/hosting-menu`)|
|-|-|
|Unlaunched site<br><br><img width="797" alt="image" src="https://github.com/user-attachments/assets/7b64c33e-1248-4fe1-bd1c-2eeb8a9174f8">|<img width="746" alt="image" src="https://github.com/user-attachments/assets/3e515d47-37e9-4664-9ebb-95751bf71f3a">|
|Launched site<br><br><img width="798" alt="image" src="https://github.com/user-attachments/assets/97161081-0527-44e7-96ae-c3ef3161d662">|<img width="746" alt="image" src="https://github.com/user-attachments/assets/8f8940a4-55cf-4918-a125-eb59d6a8ee4b">


## Why are these changes being made?


pbxlJb-6ye-p2

## Testing Instructions



1. Prepare an unlaunched Atomic site.
    - Go to `/sites/settings/site/:siteSlug`.
    - Verify you can launch the site.
1. Prepare another unlaunched Atomic site.
    - Go to `/settings/general/:siteSlug?flags=-untangling/hosting-menu`.
    - Verify you can still launch the site.
1. Prepare any Atomic site.
    - Go to `/sites/settings/site/:siteSlug`.
    - Verify you can modify the privacy settings.
    - Go to `/settings/general/:siteSlug?flags=-untangling/hosting-menu`.
    - Verify you can still modify the privacy settings.

## Pre-merge Checklist
